src/proxy/inbound_passthrough.rs

```diff
@@ -55,7 +56,17 @@ impl InboundPassthrough {
                         }
                     });
                 }
-                Err(e) => error!("Failed TCP handshake {}", e),
+                Err(e) => {
+                    debug_assert_eq!(e.get_ref().unwrap().to_string(), ERR_TOKIO_RUNTIME_SHUTDOWN);
```

Comment:
Unless I'm missing something, this will `panic` in case of any other error, which I don't believe is something we should be doing.

Reply:
It will  not panic for release

/// Unlike [`assert_eq!`], `debug_assert_eq!` statements are only enabled in non
/// optimized builds by default. An optimized build will not execute
/// `debug_assert_eq!` statements unless `-C debug-assertions` is passed to the
/// compiler. This makes `debug_assert_eq!` useful for checks that are too
/// expensive to be present in a release build but may be helpful during
/// development. The result of expanding `debug_assert_eq!` is always type checked.